src/python/pants/backend/project_info/peek.py

```diff
@@ -126,27 +150,57 @@ def default(self, o):
             return str(o)
 
 
+@rule
+async def get_target_data(targets: UnexpandedTargets) -> TargetDatas:
```

Comment:
As discussed, in a followup, I think this should be `Targets`. Good for now to stick to what we had before.